src/modules/m_nick.c

```diff
@@ -294,14 +294,13 @@ void nick_identify (Nick *nptr, User *uptr __unused, char *all)
         NoticeToUser(nptr,"Syntax: \2NICK IDENTIFY \037password\037\2");
         return;
     }   
-        
     user = find_user(nptr->nick);
     if (!user) {
         NoticeToUser(nptr,"You are not registered");
         return;
     }   
-        
-    if (user->authed == 1) {
+    
+    if (nptr->svid != NULL) {
```

Comment:
This will always be true, given that svid is an array. What you want instead is:

if (*nptr->svid)
or
if (nptr->svid[0])

Ideally, this should be offloaded to a function returning a boolean, but I guess we can leave that for a later cleanup.

Reply:
Right, weirdly enough the review was done on an old version, this is already fixed in the branch: https://github.com/xiu/child/blob/svid/src/modules/m_nick.c#L302. It's slightly different than your second option though, I'll come up with a function.